src/test/java/cloud/prefab/client/integration/DynamicTestBuilder.java

```diff
@@ -0,0 +1,76 @@
+package cloud.prefab.client.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import cloud.prefab.client.Options;
+import cloud.prefab.client.PrefabCloudClient;
+import cloud.prefab.client.integration.Expectation;
+import cloud.prefab.client.integration.Scenario;
+import cloud.prefab.domain.Prefab;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.function.Executable;
+
+public class DynamicTestBuilder {
+
+  public static DynamicTest build(
+    Map<String, Object> testData,
+    String version,
+    String function
+  ) {
+    Object input = testData.get("input");
+    Object expected = testData.get("expected");
+
+    String testCase = String.format(
+      "%s : %s : %s",
+      version,
+      function,
+      testData.get("name")
+    );
+
+    Executable exec = () -> {
+      System.out.println(testCase);
+
+      Expectation expectation = new Expectation();
+      Scenario scenario;
+
+      if (input instanceof String) {
+        scenario =
+          new Scenario(
+            function,
+            "config_client",
+            (String) input,
+            (Map<String, Object>) testData.get("client_overrides")
+          );
+      } else {
+        scenario =
+          new Scenario(
+            function,
+            "config_client",
+            (Map<String, Object>) input,
+            (Map<String, Object>) testData.get("client_overrides")
+          );
+      }
+
+      if (expected instanceof String) {
```

Comment:
`Expection expectaion = new Expecation(expected)` and constructors should sort this out.

Reply:
huzzah!